keeper/component/dispatcher/dispatcher.go

```diff
@@ -124,32 +127,30 @@ func (d *Dispatcher) DispatchPayload(
 	transferAttr *types.TransferAttributes,
 	payload *core.Payload,
 ) error {
-	if err := d.validatePayload(payload); err != nil {
+	if err := d.ValidatePayload(payload); err != nil {
+		// TODO: look at how it is displayed
 		return core.ErrValidation.Wrap(err.Error())
 	}
 
 	if err := d.dispatchActions(ctx, transferAttr, payload.PreActions); err != nil {
-		return fmt.Errorf("actions dispatch failed: %w", err)
+		return errors.Wrap(err, "actions dispatch failed")
 	}
 
 	if err := d.dispatchForwarding(ctx, transferAttr, payload.Forwarding); err != nil {
-		return fmt.Errorf("forwarding dispatch failed: %w", err)
+		return errors.Wrap(err, "forwarding dispatch failed")
```

Comment:
nit I think it would be nice to always add the import alias `errorsmod` here to have it consistent and avoid conflicts with the builtin `errors` package from the standard library